### PR TITLE
doc[compatibility]:provide compatibility doc

### PR DESCRIPTION
**Pull request by steven-zou**
- list all the supported replication adapters
- list all the verified OIDC providers including the ones verified by the end users
- list all the pluggable scanners

Signed-off-by: Steven Zou <szou@vmware.com>

### CODE REVIEW

**Comment by steven-zou** [comment]
@xaleeks 

Any comments?

File: docs/support_metrics.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Support Metrics
```
Comment (xaleeks) [inline docs/support_metrics.md]: I would call it 'Harbor compatibility list'

File: docs/support_metrics.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Here defines all the support metrics info of Harbor.
```
Comment (xaleeks) [inline docs/support_metrics.md]: Compatibility listed by components

File: docs/support_metrics.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+| docker hub | ![docker hub](img/replication_adapters/docker_hub.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)|
+| GCR | ![GCR](img/replication_adapters/gcr.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)|
+| ECR | ![ECR](img/replication_adapters/ecr.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)|
+| ACR | ![ACR](img/replication_adapters/acr.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)|
+| AliCR | ![AliCR](img/replication_adapters/ali-cr.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)|
+| Helm Hub | ![Helm Hub](img/replication_adapters/helm-hub.png)|![Y](img/replication_adapters/right.png)| N/A |
+| Huawei SWR | ![Huawei SWR](img/replication_adapters/hw.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)|
+
+**Notes**: Pull mode means pull content from the specified source registries into harbor. Push mode means push the content of Harbor to the specified target registries.
```
Comment (xaleeks) [inline docs/support_metrics.md]: minor adjustments -> 'Pull mode is replicating artifacts from specified source registries into Harbor.  Push mode is replicating artifacts from Harbor to specified target registries'

**Comment by steven-zou** [comment]
@stuclem 

Will you please review this document? Thanks!

File: docs/support_metrics.md (hunk 1 above)
Comment (stuclem) [reply docs/support_metrics.md]: Harbor Compatibility List

File: docs/support_metrics.md (hunk 2 above)
Comment (stuclem) [reply docs/support_metrics.md]: This document provides compatibility information for all Harbor components.

**Comment by stuclem** [comment]
> @stuclem
> 
> Will you please review this document? Thanks!

@steven-zou done!

**Comment by reasonerjt** [comment]
I'm approving and I think we need to update it in 1.10 branch?

**Comment by steven-zou** [comment]
@stuclem @xaleeks 

Doc updated, please take a look!

File: docs/harbor_compatibility_list.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+| Helm Hub | ![Helm Hub](img/replication_adapters/helm-hub.png)|![Y](img/replication_adapters/right.png)| N/A | V1.9 | N |
+| Artifactory | ![Artifactory](img/replication_adapters/artifactory.png)|![Y](img/replication_adapters/right.png)| ![Y](img/replication_adapters/right.png) | V1.10 | N |
+| Quay | ![Quay](img/replication_adapters/quay.png)|![Y](img/replication_adapters/right.png)| ![Y](img/replication_adapters/right.png) | V1.10 | N |
+| GitLab Registry | ![GitLab Registry](img/replication_adapters/gitlab.png)|![Y](img/replication_adapters/right.png)| ![Y](img/replication_adapters/right.png) | V1.10 | N |
+
+**Notes**: 
+
+* `Pull` mode is replicating artifacts from specified source registries into Harbor. 
+* `Push` mode is replicating artifacts from Harbor to specified target registries.
+
```
Comment (stuclem) [inline docs/harbor_compatibility_list.md]: * `Pull` mode replicates artifacts from the specified source registries into Harbor. 
* `Push` mode replicates artifacts from Harbor to the specified target registries.

Comment (stuclem) [reply docs/harbor_compatibility_list.md]: Maybe also add a line here to explain what "Automated Pipeline Covered" means, unless you think that this is obvious to users.

File: docs/harbor_compatibility_list.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+## OIDC Adapter
```
Comment (stuclem) [inline docs/harbor_compatibility_list.md]: OIDC Adapters

File: docs/harbor_compatibility_list.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+|   |  OIDC Providers | Officially Verified | End User Verified   | Verified in Release |
+|---|-----------------|---------------------|---------------------|-----------------------|
+| Google Identity | ![google identity](img/OIDC/google_identity.png)| ![Y](img/replication_adapters/right.png) |  |V1.9|
+| Dex | ![dex](img/OIDC/dex.png) | ![Y](img/replication_adapters/right.png)| | V1.9 |
+| Ping Identity | ![ping identity](img/OIDC/ping.png) | | ![Y](img/replication_adapters/right.png)| V1.9 |
+| Keycloak | ![Keycloak](img/OIDC/keycloak.png) | ![Y](img/replication_adapters/right.png) | | V1.10 |
+| Auth0 | ![Auth0](img/OIDC/auth0.png) | ![Y](img/replication_adapters/right.png) | | V1.10 |
+
+## Scanner Adapter
```
Comment (stuclem) [inline docs/harbor_compatibility_list.md]: Scanner Adapters

File: docs/harbor_compatibility_list.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+|---|----------|-----------|-----------|------------|--------------------|
+| [Clair](https://github.com/goharbor/harbor-scanner-clair)    |![Clair](img/scanners/clair.png)| CentOS    |![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)| v1.10 |
+| [Anchore](https://github.com/anchore/harbor-scanner-adapter) |![Anchore](img/scanners/anchore.png)   | Anchore    |![Y](img/replication_adapters/right.png)| N | v1.10 |
+| [Trivy](https://github.com/aquasecurity/harbor-scanner-trivy)|![Trivy](img/scanners/trivy.png)| Aqua    |![Y](img/replication_adapters/right.png)| N | v1.10 |
+| [CSP](https://github.com/aquasecurity/harbor-scanner-aqua)   |![Aqua](img/scanners/aqua.png)| Aqua    | N | N | v1.10 |
+| [DoSec](https://github.com/dosec-cn/harbor-scanner/blob/master/README_en.md)|![DoSec](img/scanners/dosec.png)    | DoSec    | N | N | v1.10 |
+
+**Notes:**
+
+* `Evaluated` means the scanner implementation has been officially tested and verified.
```
Comment (stuclem) [inline docs/harbor_compatibility_list.md]: `Evaluated` means that the scanner implementation has been officially tested and verified.

File: docs/harbor_compatibility_list.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+* `As Default` means the scanner is selected as default option and can be deployed together with Harbor main components by providing extra options.
```
Comment (stuclem) [inline docs/harbor_compatibility_list.md]: `As Default` means that the scanner is provided as a default option and can be deployed together with the main Harbor components by providing extra options during installation. You must install other scanners manually.

File: docs/harbor_compatibility_list.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+| ECR | ![ECR](img/replication_adapters/ecr.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)| V1.9 | Y |
+| ACR | ![ACR](img/replication_adapters/acr.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)| V1.9 | N |
+| AliCR | ![AliCR](img/replication_adapters/ali-cr.png)|![Y](img/replication_adapters/right.png)|![Y](img/replication_adapters/right.png)| V1.9 | N |
+| Helm Hub | ![Helm Hub](img/replication_adapters/helm-hub.png)|![Y](img/replication_adapters/right.png)| N/A | V1.9 | N |
+| Artifactory | ![Artifactory](img/replication_adapters/artifactory.png)|![Y](img/replication_adapters/right.png)| ![Y](img/replication_adapters/right.png) | V1.10 | N |
+| Quay | ![Quay](img/replication_adapters/quay.png)|![Y](img/replication_adapters/right.png)| ![Y](img/replication_adapters/right.png) | V1.10 | N |
+| GitLab Registry | ![GitLab Registry](img/replication_adapters/gitlab.png)|![Y](img/replication_adapters/right.png)| ![Y](img/replication_adapters/right.png) | V1.10 | N |
+
```
Comment (stuclem) [inline docs/harbor_compatibility_list.md]: Could we provide links in the first column?

File: docs/harbor_compatibility_list.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+## OIDC Adapter
+
+|   |  OIDC Providers | Officially Verified | End User Verified   | Verified in Release |
+|---|-----------------|---------------------|---------------------|-----------------------|
+| Google Identity | ![google identity](img/OIDC/google_identity.png)| ![Y](img/replication_adapters/right.png) |  |V1.9|
+| Dex | ![dex](img/OIDC/dex.png) | ![Y](img/replication_adapters/right.png)| | V1.9 |
+| Ping Identity | ![ping identity](img/OIDC/ping.png) | | ![Y](img/replication_adapters/right.png)| V1.9 |
+| Keycloak | ![Keycloak](img/OIDC/keycloak.png) | ![Y](img/replication_adapters/right.png) | | V1.10 |
+| Auth0 | ![Auth0](img/OIDC/auth0.png) | ![Y](img/replication_adapters/right.png) | | V1.10 |
+
```
Comment (stuclem) [inline docs/harbor_compatibility_list.md]: Could we provide links in the first column?

File: docs/harbor_compatibility_list.md (hunk 4 above)
Comment (steven-zou) [reply docs/harbor_compatibility_list.md]: I think it's obvious. Let's keep it as it is. If there is confusion caused in the future, then we can add it.

File: docs/harbor_compatibility_list.md (hunk 9 above)
Comment (steven-zou) [reply docs/harbor_compatibility_list.md]: added

File: docs/harbor_compatibility_list.md (hunk 10 above)
Comment (steven-zou) [reply docs/harbor_compatibility_list.md]: added

**Comment by steven-zou** [comment]
@stuclem 

doc updated. If everything is ok, I'll merge it soon.

**Pull request merged**